### DISCUSSION TITLE
Want the page to refresh every 15 seconds but it is refreshing uncontrollably

### DISCUSSION THREAD

**Discussion by dega5** [zauberzeug/nicegui#1974] [Q&A] [answered]
### Question

`ui.timer(15, lambda: ui.open('/task-list'))`

I have this line of code inside the task-list page. When I run this, it just refreshes the page uncontrollably. What am I doing wrong?

Thanks!

**Comment by falkoschindler** [comment, marked as answer]
The timer will immediately call `ui.open` and wait for 15 seconds afterwards before repeating the call again and again.

But you probably want to call `ui.open` _once_ after 15 seconds. This can be done using `ui.timer(..., once=True)`.

**Comment by v479038280** [comment]
> ### Question
> `ui.timer(15, lambda: ui.open('/task-list'))`
> 
> I have this line of code inside the task-list page. When I run this, it just refreshes the page uncontrollably. What am I doing wrong?
> 
> Thanks!

The timer works correctly every 15 seconds ,I test it ,May be there is some thing wrong with other code

**Comment by dega5** [comment]
Thank you soo much!!
